### PR TITLE
Support Framework ScaleUp/ScaleDown with Strong Safety Guarantee

### PR DESCRIPTION
**Pull request by yqwang-ms**
See https://github.com/microsoft/frameworkcontroller/blob/yqwang/dev/doc/user-manual.md#FrameworkRescale

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,33 +5,42 @@
...
 
 These kinds of applications include but not limited to:
-- __Stateless and Stateful Service__ (Nginx, TensorFlow Serving, HBase, Kafka, etc)
-- __Stateless and Stateful Batch__ (KD-Tree Building, Batch Data Processing, etc)
-- __Any combination of above applications__ (Distributed TensorFlow Training, Stream Data Processing, etc)
+- __Stateless and Stateful Service__:
+  - DeepLearning Serving: [TensorFlow Serving](https://www.tensorflow.org/tfx/guide/serving), etc.
+  - Big Data Serving: HDFS, HBase, Kafka, Nginx, etc.
+- __Stateless and Stateful Batch__:
+  - DeepLearning MPI Training: [Horovod Training](https://horovod.readthedocs.io), etc.
```
Comment (abuccts) [inline README.md:15]:
```suggestion
  - DeepLearning MPI Training: [Horovod Distributed Training](https://horovod.readthedocs.io/en/stable/mpirun.html), etc.
```

File: README.md [hunk 2]
```diff
@@ -5,33 +5,42 @@
...
+  - DeepLearning Elastic Training without Server: [PyTorch Elastic Training with shared etcd](https://pytorch.org/elastic), etc.
```
Comment (abuccts) [inline README.md:16]: does "shared etcd" mean using k8s etcd? it's not easy to understand

File: README.md [hunk 3]
```diff
@@ -5,33 +5,42 @@
...
+  - DeepLearning Batch/Offline Inference: [TensorFlow Inference](https://www.tensorflow.org/lite/guide/inference), etc.
```
Comment (abuccts) [inline README.md:17]: tf lite is used for mobile/edge devices, maybe you can change another link

File: README.md (hunk 1 above)
Comment (yqwang-ms) [reply README.md:15]: OK

File: README.md (hunk 2 above)
Comment (yqwang-ms) [reply README.md:16]: The "shared etcd" is from https://pytorch.org/elastic/0.2.0/examples.html#multi-node .
Any suggestion to better short name?

File: README.md (hunk 3 above)
Comment (yqwang-ms) [reply README.md:17]: I cannot find doc for tf Inference, do you have one? Thanks!

File: README.md (hunk 1 above)
Comment (yqwang-ms) [reply README.md:15]: BTW, does all Horovod Training can be considered as MPI based Training (no need param server, and only need a single work task role)?

File: README.md (hunk 2 above)
Comment (yqwang-ms) [reply README.md:16]: how about "with whole cluster shared etcd" and "with per-application dedicated etcd"?

File: README.md (hunk 1 above)
Comment (abuccts) [reply README.md:15]: no, there's single GPU training, and besides mpi for distributed training, Horovod also supports other communication library like [Gloo](https://github.com/horovod/horovod#gloo)

File: README.md (hunk 2 above)
Comment (abuccts) [reply README.md:16]: that's fine, I didn't find that in pytorch docs

File: README.md (hunk 3 above)
Comment (abuccts) [reply README.md:17]: maybe [PyTorch inference](https://pytorch.org/tutorials/recipes/recipes/saving_and_loading_models_for_inference.html)?

Comment (yqwang-ms) [reply README.md:17]: Great, Thanks!

**Comment by xudifsd** [comment]
It would be great if this project has some tests.

**Comment by yqwang-ms** [comment]
> It would be great if this project has some tests.

Sure! We should add some tests later

**Pull request merged**